### ISSUE TITLE
anti-hacking system detected modification of the miner memory

### ISSUE THREAD

**Issue by raven0931** [develsoftware/GMinerRelease#135]
Hello i am new on Gminer (RivenCoin) miner is giving this error:

anti-hacking system detected modification of the miner memory

i am tried new and old nvidia driver. I dont have Antivirus program.

How can i fix it?

**Comment by senski7** [comment]
I have the same problen!

"19:14:42 Anti-hacking system detected modification of the miner memory. If you using an antivirus, add miner to antivirus exceptions
19:14:43 Miner terminated, watchdog will restart process after 10 seconds"

What can i do ?

**Comment by Dasolo** [comment]
Hello. 
I have the same problem when I try to start GMiner from .net core app. 
Can I fix it or should I wait for new release?

**Comment by eBuccaneer** [comment]
Hi,

same problem here when running it from a golang executable managing my mining services.
Running on ubuntu 18.04 with definitely no anti-virus software installed/active.

**Comment by gabriel-milan** [comment]
Same here, even with v2.46 that have just been launched

**Comment by imakin** [comment]
v2.47 same issue here, i'm running on linux with nvidia, there's no way i have anti-virusminer,

i can mine with no problem with other miner 

it says fixed in v1.52 almost 2 years ago, but still occured now
https://github.com/develsoftware/GMinerRelease/issues/23

**Comment by develsoftware** [comment]
may be you inject miner with LD_PRELOAD, check this environment variable

**Comment by imakin** [comment]
LD_PRELOAD is libgtk3-nocsd.so.0
what am i suppose to change?

**Comment by recolic** [comment]
Please disable this bug on linux. there's no anti-virus software at all, but still reporting this bug.

**Comment by develsoftware** [comment]
it isn't bug, it's way to .so injection to hack miner

**Comment by AaronMcDaniel** [comment]
Bug or not this program does not work as described on linux. I am on ubuntu 18.04, with an RTX 2080s and I get this issue when I run the miner script. Is there additional setup required for linux? `export LD_PRELOAD=` does nothing since it doesn't assign a value to that variable. Is there a value this variable is supposed to have? If so that should either be added to the README or to the error statement

**Comment by recolic** [comment]
I must have the freedom to be injected, just like that I should have the freedom to shoot my foot. that's why we're on linux. 

this feature is ridiculous even if it doesn't cause a problem.  there must be a easy switch to turn it off by design.

**Comment by Dasolo** [comment]
`export LD_PRELOAD=` did not work for me. I try start miner from .net app and still get same error.

**Comment by ketchmeup** [comment]
Same here. Ubuntu 18 and LD_PRELOAD= did not work for me too. Is there a way to disable this feature? I am executing your binary with Python/pexpect  on different systems. No virus no anti-virus etc.
Thanks!

**Comment by holzerseb** [comment]
Same issue, cannot start from my .net app

**Comment by imakin** [comment]
i think you can also ssh to yourself@localhost, use sudo, or fireup `tmux`

**Comment by ketchmeup** [comment]
@holzerseb 
I use openssh on windows to connect from my unix environment (with pexpect/python) and start gpu mining with a .bat file.
Have you tried to put commands in a .bat file and executing the .bat file from .net?

**Comment by holzerseb** [comment]
@ketchmeup 
yeah I've tried starting a bat file as well as calling it directly. I have found no way to start it via .net

**Comment by ketchmeup** [comment]
@holzerseb
Do you need control over the mining process? If not, maybe you could start the miner in a new proces/cmd (start.exe or cmd.exe?). I am not very familiar with process and env handling on windows. For me it works if I connect to my openssh on windows and execute the .bat or .cmd file.

**Comment by Qbuer** [comment]
I tried to use the proxychains-ng to use gminer behinded  firewall and got this error message.

**Comment by lucidyan** [comment]
Running miner inside the tmux solved this error for me

**Comment by LiveWire1968** [comment]
on linux, change the script #!/bin/sh to #!/bin/bash.

**Comment by g00dnatur3** [comment]
hey bub,
because you cannot relax this constraint for us linux users - i have switched to bminer

gminer will not allow me to run via a programatic child process
bminer does

you lost a customer....

**Comment by DrNopt** [comment]
Use latest build 2.60. It's fixed !
Windows 11 - 22000.65

**Comment by eXPeRi91** [comment]
> 
> 
> Use latest build 2.60. It's fixed !
> Windows 11 - 22000.65

"Edition	Windows 11 Pro
Update	21H2
Installed on	‎7/‎12/‎2021
OS build	22000.65
Experience	Windows Feature Experience Pack 421.17400.45.3
"

Not fixed.

**Comment by matholio** [comment]
v2.60 on W10 21H1, starting miner with Power Automate triggers antihacking.
Is there an GMiner option to disable this feature?

**Comment by bessgeor** [comment]
2.54 for those of us using .NET, try to run without a debugger (Ctrl+F5 in VS by default)

**Comment by LiveWire1968** [comment]
Kind of screwed up that gminers "antihacking" required linux users to use sudo to work. I backed away in 30 seconds. It has nothing over the other mining software to worry about.

**Comment by dhruvhaldar** [comment]
> on linux, change the script #!/bin/sh to #!/bin/bash.

Working on Linux Ubuntu 21.04

**Comment by zee00100** [comment]
> > on linux, change the script #!/bin/sh to #!/bin/bash.
> 
> Working on Linux Ubuntu 21.04

Works for me also on Manjaro

**Comment by imakin** [comment]
With lots of misconception, the real problem is the error message.

Instead of showing **"antihacking"** word, just change the message to **"clear LD_PRELOAD environment variable before using this program"**

**Comment by bonlime** [comment]
This is a very unpleasant behaviour. Encountered it when tried launching GMiner from python script. Default `subprocess.run` doesn't work `subprocess.Popen` works, but if I try to call the same command from within loop - it doesn't work. On some machines launching inside tmux works, in others - doesn't. Option to remove anti-hack check (or a cleaner explanation how to bypass it) would be nice.

**Comment by NiclasEriksen** [comment]
> on linux, change the script #!/bin/sh to #!/bin/bash.

More precisely, this seem to have to match the shell you're using. In my case it was `zsh`. I get the frustration in some of these comments, but please understand that miner software is a prime target for rootkits and exploits and I personally appreciate the caution.

**Comment by kirby561** [comment]
I just ran into this using gminer 2.75 on Windows 10. I was able to work around it by running miner.exe from git bash. I'm not sure why it wasn't working to begin with. Clearing LD_PRELOAD did not work for me.

**Comment by mthqwork** [comment]
Had the same issue with fish shell, on Arch Linux. Even if script specified #!/bin/bash or #!/bin/sh shebang.
Running bash, then running the miner worked fine however.

**Comment by silence48** [comment]
![image](https://user-images.githubusercontent.com/35656692/163887269-c009f4d6-a8e9-4103-8149-0d378b5bff15.png)
![image](https://user-images.githubusercontent.com/35656692/163887353-30672180-5749-4093-9924-ddd90afe4bb3.png)

**Comment by liwengithub** [comment]
really stupid,Anti-hacking
